analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -74,7 +74,12 @@ public void Execute()
                 return;
             }
             var current = queue.Dequeue();
-            if (visited.Add(current) && current.AddVisit() <= MaxOperationVisits)
+            if (visited.Add(current)
+                && current.AddVisit() is var visitCount
+                && (visitCount <= MaxOperationVisits
+                || (visitCount == MaxOperationVisits + 1
+                    && current.Block.ConditionalSuccessor is not null
+                    && syntaxClassifier.IsInLoopCondition(current.Block.BranchValue?.Syntax))))
```

Comment:
This combo of && || && got too big for begin in the main SE loop. Let's extract it into a method `CheckVisitCount` or something like that.

It's not going to be an arrow method to make it more readable. I'd format it as
..<=..
|| (.. <= +1 && && )